clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -516,8 +527,9 @@ private void fetchOffsetsWithRetries(final OffsetFetchRequestState fetchRequest,
                 result.complete(res);
             } else {
                 if (error instanceof RetriableException || isStaleEpochErrorAndValidEpochAvailable(error)) {
-                    if (error instanceof TimeoutException && fetchRequest.isExpired) {
-                        result.completeExceptionally(error);
+                    if (fetchRequest.isExpired()) {
+                        log.debug("Fetch request for {} timed out and won't be retried anymore", fetchRequest.requestedPartitions);
```

Comment:
nit: let's use "OffsetFetch request" to avoid confusion with Fetch requests

Reply:
Done.